apps/src/pythonlab/pythonHelpers/messageHelpers.ts

```diff
@@ -41,12 +55,57 @@ export function parseErrorMessage(errorMessage: string) {
   }
   if (mainErrorLine >= errorLines.length) {
     // If we never find the main.py error, return the entire message.
-    return errorMessage;
+    return neighborhoodExceptionMessage || errorMessage;
+  }
+  if (neighborhoodExceptionMessage) {
+    // Include the line number and call in main.py that resulted in Neighborhood exception.
+    return `${neighborhoodExceptionMessage}\n${errorLines
```

Comment:
Does the error parsing still work if the error was thrown from a file other than main?

Reply:
https://github.com/code-dot-org/code-dot-org/blob/cc62d0f0a529b782e93456255512e4b1d94795df/apps/test/unit/pythonlab/pythonHelpers/messageHelpersTest.ts#L73

Reply:
Noting that if the actual cause of the exception is in a student's helper file (e.g. a call to a helper function), then the details include the call to the helper function in `main.py`. 

https://github.com/user-attachments/assets/962eaef0-9249-4bbc-82b6-dcd702902fc9

Reply:
Checked with curriculum and will include entire parsed error message for easier debugging. [Slack thread](https://codedotorg.slack.com/archives/C06JELCAPT9/p1738094478925619?thread_ts=1738006221.572969&cid=C06JELCAPT9)